plugins/clipboard/plugin.js

```diff
@@ -502,6 +569,32 @@
 					};
 				} );
 			}
+
+			// Detect if any of paste buttons was touched. In such case we assume that user is using
+			// touch device and force displaying paste dialog (#595).
+			if ( editor.ui.addButton ) {
+				// Waiting for editor instance to be ready seems to be the most reliable way to
+				// be sure that paste buttons are already created.
+				editor.once( 'instanceReady', function() {
+					if ( !editor._.pasteButtons ) {
+						return;
+					}
+
+					CKEDITOR.tools.array.forEach( editor._.pasteButtons, function( name ) {
+						var pasteButton = editor.ui.get( name );
+						// Check if button was not removed by `removeButtons` config.
+						if ( pasteButton ) {
+							var buttonElement = CKEDITOR.document.getById( pasteButton._.id );
+
+							buttonElement.on( 'touchend', function() {
```

Comment:
I was thinking about optimizing this listener, and putting it once on the entire toolbar and just then check what button was pressed. However after playing a bit with it I start to think that it's too much of effort for the benefit. So I guess it's fine the way it is.